script/index.ts

```diff
@@ -52,39 +90,75 @@ function buildOptionsForLanguage(language: Language, beautifiers: String[]) {
       enum: option.enum,
       minimum: option.minimum,
       maximum: option.maximum,
-      items: option.items
+      items: option.items,
     };
-  })
-  languageOptions["beautifiers"] = {
+  });
+  languageOptions.beautifiers = {
     title: "Beautifiers",
     type: "array",
     default: beautifiers,
-    description: `Comma separated list of beautifiers to apply when beautifying ${language.name} code`,
+    description: `Comma separated list of beautifiers to apply when beautifying ${
+      language.name
+    } code`,
     items: {
-      type: "string"
-    }
-  }
-  languageOptions["beautify_on_save"] = {
+      type: "string",
+    },
+  };
+  languageOptions.beautify_on_save = {
     title: `Beautify ${language.name} On Save`,
     type: "boolean",
     default: false,
     description: `Automatically beautify ${language.name} files on save`,
-    order: -1
-  }
+    order: -1,
+  };
   return languageOptions;
 }
 
-function writeOptionsJson() {
-  const languageOptions = buildOptions();
-  let optionsString = JSON.stringify(languageOptions, null, 2);
-  let outputFile = path.resolve(__dirname, '../dist/options.json');
-  fs.writeFile(outputFile, optionsString, (error: Error) => {
-    if (error) {
-      throw error;
-    }
+function buildBeautifierOptions() {
+  const beautifierOptions: AtomConfigRegistry = {};
+  beautifiers.forEach((beautifier, index) => {
+    const beautifierProperties = buildOptionsForBeautifier(beautifier);
+    beautifierOptions[beautifier.name] = {
+      title: beautifier.name,
+      description: `Configure ${beautifier.name}`,
+      type: "object",
+      collapsed: true,
+      order: index,
+      properties: beautifierProperties,
+    };
   });
+  return beautifierOptions;
 }
 
+function buildOptionsForBeautifier(beautifier: Beautifier) {
+  const beautifierOptions: AtomConfigRegistry = {};
+  if (!!beautifier.resolveConfig) {
```

Comment:
```suggestion
  if (beautifier.resolveConfig) {
```

Reply:
It doesn't return a boolean though, just Promise<ResolvedConfig>.

Reply:
It's still truthy: https://developer.mozilla.org/en-US/docs/Glossary/truthy 

```
if (!!beautifier.resolveConfig) {
```

and

```
if (beautifier.resolveConfig) {
```

are effectively the same.